Jenkinsfile

```diff
@@ -1,101 +1,115 @@
-        throw t;
-    }
-    finally {
-        if (tearDown) {
-            tearDown();
-        }
-    }
-}
 
+pipeline {
+    agent any
 
-node {
-    stage("Checkout") {
-        checkout scm
+    environment {
+        SHORT_UUID = sh( script: "uuidgen | cut -d '-' -f1", returnStdout: true).trim()
+        COMPOSE_PROJECT_NAME = "${PROJECT_NAME}-${env.SHORT_UUID}"
+        VERSION = env.BRANCH_NAME.replace('/', '-').toLowerCase().replace(
+            'master', 'latest'
+        )
+        IS_RELEASE = "${env.BRANCH_NAME ==~ "release/.*"}"
     }
 
-    stage('Test') {
-        tryStep "test", {
-            sh "api/deploy/test/test.sh"
+    stages {
+        stage('Test') {
+            steps {
+                sh 'make test'
+            }
         }
-    }
 
-    stage("Build dockers") {
-        tryStep "build", {
-            docker.withRegistry("${DOCKER_REGISTRY_HOST}",'docker_registry_auth') {
-                def api = docker.build("datapunt/waarnemingen-mensen:${env.BUILD_NUMBER}", "--build-arg AUTHORIZATION_TOKEN=dev --build-arg GET_AUTHORIZATION_TOKEN=get-auth-token --build-arg SECRET_KEY=dev api")
-                api.push()
-                api.push("acceptance")
+        stage('Build') {
+            steps {
+                sh 'make build'
             }
         }
-    }
-
-    stage("Locust load test") {
-        sh("./api/deploy/docker-locust-load-test.sh")
-    }
-}
-
-String BRANCH = "${env.BRANCH_NAME}"
-
-if (BRANCH == "master") {
 
-    node {
-        stage('Push acceptance image') {
-            tryStep "image tagging", {
-               docker.withRegistry("${DOCKER_REGISTRY_HOST}",'docker_registry_auth') {
-                    def image = docker.image("datapunt/waarnemingen-mensen:${env.BUILD_NUMBER}")
-                    image.pull()
-                    image.push("acceptance")
+        stage('Push and deploy') {
+            when {
+                anyOf {
+                    branch 'master'
+                    buildingTag()
+                    environment name: 'IS_RELEASE', value: 'true'
                 }
             }
-        }
-    }
+            stages {
+                stage('Push') {
+                    steps {
+                        retry(3) {
+                            sh 'make push'
+                        }
+                    }
+                }
 
-    node {
-        stage("Deploy to ACC") {
-            tryStep "deployment", {
-                build job: 'Subtask_Openstack_Playbook',
-                parameters: [
-                    [$class: 'StringParameterValue', name: 'INVENTORY', value: 'acceptance'],
-                    [$class: 'StringParameterValue', name: 'PLAYBOOK', value: 'deploy-waarnemingen-mensen.yml'],
-                ]
-            }
-        }
-    }
+                stage('Deploy to acceptance') {
+                    when {
+                        anyOf {
+                            environment name: 'IS_RELEASE', value: 'true'
```

Comment:
Deze wil je misschien uitbreiden met `buildingTag()`. Op die manier gaan alle prod releases altijd eerst naar acceptance. Hebben we niet overal, maar kan handig zijn. Persoonlijke voorkeur.

Reply:
`buildingTag()` staat er al boven op regel 44